store/src/main/java/org/apache/rocketmq/store/CommitLog.java

```diff
@@ -1634,12 +1634,18 @@ private void swapRequests() {
         private void doCommit() {
             if (!this.requestsRead.isEmpty()) {
                 for (GroupCommitRequest req : this.requestsRead) {
-                    // There may be a message in the next file, so a maximum of
-                    // two times the flush
-                    boolean flushOK = CommitLog.this.mappedFileQueue.getFlushedWhere() >= req.getNextOffset();
-                    for (int i = 0; i < 2 && !flushOK; i++) {
+                    boolean flushOK = false;
```

Comment:
How about flushOK=CommitLog.this.mappedFileQueue.getFlushedWhere() >= req.getNextOffset() when init？ Prevent every request from flushing at least once.

Reply:
OK